### PR TITLE
Update behavior for disabled text area element

### LINKED ISSUE #511: Bug/irregularity: Form element [Read-only text area] does not act like a [read-only] element
A small bug/irregularity report about form element `[Read-only text area]`, that does not act like a `[read-only]` element
Grafana 11.2.2, Business Forms 4.6.0 

 **[Read-only text area]**
- Values color blue (or any Highlight Changes color) on data changes, not grey like a [read-only] element (after context.panel.patchFormValue);
```
  context.panel.patchFormValue({
    'Somefield1': '2024-10-04',
    'Somefield2': '12:00',
    'Description': 'Some text',
    'LastActivityNotes': 'Some other text'
  });
```
- On clicking submit, the Update Confirmation Window (with display values > Updated only) shows the [Read-only text area] values as changes.

```
      // ---Current quick fix:---
      let currentdata = context.panel.initial;  // First read the whole dataset, since there is no patchtInitial
      currentdata.Description = jsonpayload.Description;  // [Read-only text area 1]
      currentdata.LastActivityNotes = jsonpayload.LastActivity?.Notes;  // [Read-only text area 2]
      context.panel.setInitial(currentdata);
```

It would be nice if the `[Read-only text area]` would act like a `[read-only]` element.

Thanks!
Best regards, Paul

### PR DESCRIPTION
**Pull request by vitPinchuk**
Resolves #511

### CODE REVIEW

File: provisioning/dashboards/datasource.json [hunk 1]
```diff
@@ -167,6 +167,350 @@
...
+        },
+        "sync": true,
+        "update": {
+          "code": "if (context.panel.response && context.panel.response.state === 'Done') {\n  context.grafana.notifySuccess(['Update', 'Values updated successfully.']);\n  context.panel.initialRequest();\n} else {\n  context.grafana.notifyError(['Update', 'An error occurred updating values.']);\n}",
+          "confirm": true,
+          "contentType": "application/json",
+          "datasource": "PostgreSQL",
+          "getPayload": "const payload = {};\n\ncontext.panel.elements.forEach((element) => {\n  if (!element.value) {\n    return;\n  }\n\n  payload[element.id] = element.value;\n})\n\n/**\n * Data Source payload\n */\nreturn payload;",
+          "method": "datasource",
+          "payload": {
```
Comment (asimonok) [inline provisioning/dashboards/datasource.json:316]: I think it should be normalized to an empty object

**Review by asimonok — CHANGES_REQUESTED** [review]

File: provisioning/dashboards/datasource.json (hunk 1 above)
Comment (vitPinchuk) [reply provisioning/dashboards/datasource.json]: updated

**Review by asimonok — APPROVED** [review]
Looks good!

@mikhail-vl ready for review

**Review by mikhail-vl — APPROVED** [review]
Looks great!

**Pull request merged**